psi4/driver/qcdb/libmintsbasissetparser.py

```diff
@@ -246,7 +255,20 @@ def parse(self, symbol, dataset):
                                 scale = float(what.group(3))
 
                                 if len(shell_type) == 1 or (len(shell_type) >= 3 and shell_type[:2] == 'L='):
-                                    am = shell_to_am[shell_type[0]] if len(shell_type)==1 else int(shell_type[2:])
+                                    if len(shell_type) == 1:
+                                        am = shell_to_am[shell_type[0]]
+                                        if am_convention == "HIK" or (am == -1 and shell_type[0].upper() == "K" and max_am_to_date == 6):
+                                            am_convention = "HIK"
+                                            am = shell_to_am_HIK[shell_type[0]]
+                                        if am_convention == "HIJ" or (am == -1 and shell_type[0].upper() == "J" and max_am_to_date == 6):
+                                            am_convention = "HIJ"
+                                            am = shell_to_am_HIJ[shell_type[0]]
+                                        if am_convention is None and am == -1:
+                                            raise ValidationError(f"""Gaussian94BasisSetParser::parse: angular momentum type {shell_type[0]} too high and HIJ/HIK convention uncertain. Use, for example, `L=7` notation. line {lineno}: {line}""")
+                                    else:
+                                        am = int(shell_type[2:])
+
+                                    max_am_to_date = max(max_am_to_date, am)
```

Comment:
It's perfectly legal to have the shells in arbitrary order, e.g. P, S, D, G, K, I, F...

Reply:
right. I added a few tests to show this still works (plus your weird basis ones). Only time when it won't work is if the HIK/HIJ convention hasn't been hinted beforehand. As that's the intersection of AM>7 letter-based basis set and weirdly ordered basis set, I'm not too worried.